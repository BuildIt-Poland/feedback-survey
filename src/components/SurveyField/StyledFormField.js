import styled from 'styled-components';
import { FormField } from 'grommet';

import media from '../../styles/media';
import { spacingMedium, fontMedium, fontSmall } from '../../styles/designTokens';

const StyledFormField = styled(FormField)`
  max-width: 80rem;
  div {
    ${props => (props.noBorderBottom ? 'border-bottom: none' : null)}
  }
  input {
    font-weight: 400;

    :focus {
      border-color: red;
    }
  }
  label:first-child {
    margin: 0 0 2rem 0;
    font-size: ${fontSmall};

    &:after {
      color: red;
      content: ${props => (props.required ? "'*'" : "''")};
      margin-left: 0.5rem;
      white-space: nowrap;
    }
  }

  ${media.phone`
    label:first-child {
      font-size: ${fontMedium};
    }
  `}
`;

export default StyledFormField;
