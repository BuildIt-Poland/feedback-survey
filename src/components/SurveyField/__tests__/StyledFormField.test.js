import React from 'react';
import { create } from 'react-test-renderer';

import StyledFormField from '../StyledFormField';

describe('COMPONENT - SurveyField StyledFormField', () => {
  it('renders correctly', () => {
    const component = create(<StyledFormField />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
