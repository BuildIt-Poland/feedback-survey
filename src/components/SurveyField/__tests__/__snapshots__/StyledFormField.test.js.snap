// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - SurveyField StyledFormField renders correctly 1`] = `
<div
  className="sc-bdVaJa fEguSI FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 CAsMy"
>
  <div
    className="StyledBox-sc-13pk1d4-0 gZSidW"
  />
</div>
`;
